### ISSUE TITLE
[data grid] Scroll glitch with master detail

### ISSUE THREAD

**Issue by PaltriPhema** [mui/mui-x#14906]
### Steps to reproduce

Link to live example: (required)

Steps:
1.Open detailPanel
2. Scroll around
3.Glitches!

### Current behavior

whenever i open the detail panel and try to move around, the scrolling glitches and it either goes all the way back up or keeps me from scrolling properly up and down

### Expected behavior

no problem scrolling even when detail panel is open

### Context

I'm trying to get MasterDetail inside a datagridpro.
In this MasterDetail i'm trying to load another datagridpro.
I set getDetailPanelHeight to auto. If i put a fixed value instead, the problem will happen only once: after scrolling me around once, it won't do it again

https://github.com/user-attachments/assets/8ac27558-3539-4aa6-aa5d-dd5233b882dc

I tried pasting (and using) the exact same code from https://mui.com/x/react-data-grid/master-detail/, changing the input data, but it still gives me this problem. We tried incrementing the number of rows present in the mui site and it gave us some kind of flicker there too

https://github.com/user-attachments/assets/e78bb4c4-5afa-4811-9702-0a55c33867d4

**Search keywords**: scrolling, detailpanelcontent, datagrid, master detail
**Order ID**: 93705

**Comment by KenanYusuf** [comment]
Looks like it could be related to #14726

**Comment by ebo-bpi** [comment]
We do have the same issue. would be nice to get a bit of prio on this. 
Our Order ID is: 102010

with the last update (8.10.2) some of the scroll issues are resolved, but the one with master - details still not.

**Comment by michelengelen** [comment]
Thanks for the report and the videos. This is resolved in two parts:

- The one-time glitch with a fixed `getDetailPanelHeight` was fixed by #14929 ([v7.21.0](https://github.com/mui/mui-x/releases/tag/v7.21.0)) and #16763 ([v8.0.0](https://github.com/mui/mui-x/releases/tag/v8.0.0)).
- The jumps with `getDetailPanelHeight: () => 'auto'` and a nested grid are inherent to virtualization: the grid cannot know the panel height before the content renders, so late size changes push the rows and create a jump. We investigated this in #20410 and concluded that a built-in correction would break native scrolling. The solution is to cache each panel's measured height and return it from `getDetailPanelHeight`. See the [Lazy loading detail panels with auto height](https://mui.com/x/react-data-grid/row-recipes/#lazy-loading-detail-panels-with-auto-height) recipe added in #20979.

I will close this issue with the same resolution as #20410. If the recipe does not solve your case on the latest version, comment here and we will take another look.
